CMSgov/easi-app
# EASI-1972/Add Component and Role Fields to SystemIntakeContact

# EASI-1972

## Changes and Description

- Added required component and role fields to system_intake_contacts table
- Updated GraphQL queries/mutations to use these new fields



## How to test this change

First query system intakes to find a valid system intake:
  ```graphql
query {
    systems(first: 0) {
      edges {
        node {
          id
        }
      }
    }
}
  ```

If there are none, create one to test with:
  ```graphql
mutation {
    createSystemIntake(input: { requestType: NEW, requester: { name: "Jeremy" } }) {
      id
    }
}
```

Create a system intake contact using one of the ID's returned above for systemIntakeId:
  ```graphql
mutation {
    createSystemIntakeContact(input: { euaUserId: "SGXL", systemIntakeId: "5ce0287a-29bd-4bb9-9ce3-871af8d5d7d8", role: "Supreme Overlord", component: "Customer Service" }) {
      systemIntakeContact{
        euaUserId
        systemIntakeId
        component
        role
      }
    }
}
  ```

Query system intake contacts to see that the results contain the new fields with the values you set:
```graphql
query {
  systemIntakeContacts(id: "5ce0287a-29bd-4bb9-9ce3-871af8d5d7d8") {
    systemIntakeContacts {
      euaUserId
      component
      role
      commonName
      email
    }
  }
}
```

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
